components/omega/src/ocn/VertMix.cpp

```diff
@@ -0,0 +1,297 @@
+
+   if (!vertmix->ComputeVertMixConv.Enabled) {
+      LOG_INFO("VertMix::init: Convective mixing is disabled.");
+   } else {
+      LOG_INFO("VertMix::init: Convective mixing is enabled.");
+      Err +=
+          ConvConfig.get("Diffusivity", vertmix->ComputeVertMixConv.ConvDiff);
+      CHECK_ERROR_ABORT(Err, "VertMix::init: Parameter Convective:Diffusivity "
+                             "not found in ConvConfig");
+
+      Err += ConvConfig.get("TriggerBVF",
+                            vertmix->ComputeVertMixConv.ConvTriggerBVF);
+      CHECK_ERROR_ABORT(Err, "VertMix::init: Parameter Convective:TriggerBVF "
+                             "not found in ConvConfig");
+   }
+
+   /// Get Shear from VertMixConfig
+   Config ShearConfig("Shear");
+   Err += VertMixConfig.get(ShearConfig);
+   CHECK_ERROR_ABORT(
+       Err, "VertMix::init: Shear subgroup not found in VertMixConfig");
+
+   /// Get shear diffusivity and viscosity parameters
+   Err += ShearConfig.get("Enable", vertmix->ComputeVertMixShear.Enabled);
+   CHECK_ERROR_ABORT(
+       Err, "VertMix::init: Parameter Shear:Enable not found in ShearConfig");
+
+   if (!vertmix->ComputeVertMixShear.Enabled) {
+      LOG_INFO("VertMix::init: Shear mixing is disabled.");
+   } else {
+      LOG_INFO("VertMix::init: Shear mixing is enabled.");
+      Err +=
+          ShearConfig.get("NuZero", vertmix->ComputeVertMixShear.ShearNuZero);
+      CHECK_ERROR_ABORT(
+          Err,
+          "VertMix::init: Parameter Shear:NuZero not found in ShearConfig");
+
+      Err += ShearConfig.get("Alpha", vertmix->ComputeVertMixShear.ShearAlpha);
+      CHECK_ERROR_ABORT(
+          Err, "VertMix::init: Parameter Shear:Alpha not found in ShearConfig");
+
+      Err += ShearConfig.get("Exponent",
+                             vertmix->ComputeVertMixShear.ShearExponent);
+      CHECK_ERROR_ABORT(
+          Err,
+          "VertMix::init: Parameter Shear:Exponent not found in ShearConfig");
+   }
+} // end init
+
+/// Compute diffusivity and viscosity for all cells/layers (no displacement)
+void VertMix::computeVertMix(const Array2DReal &NormalVelocity,
+                             const Array2DReal &TangentialVelocity,
+                             const Array2DReal &BruntVaisalaFreq) {
+   OMEGA_SCOPE(LocVertDiff, VertDiff); /// Create a local view for computation
+   OMEGA_SCOPE(LocVertVisc, VertVisc); /// Create a local view for computation
+   OMEGA_SCOPE(LocComputeVertMixConv,
+               ComputeVertMixConv); /// Local view for PP VertMix computation
+   OMEGA_SCOPE(LocComputeVertMixShear,
+               ComputeVertMixShear); /// Local view for PP VertMix computation
+
+   /// Initialize VertDiff and VertVisc to background values
+   deepCopy(LocVertDiff, BackDiff);
+   deepCopy(LocVertVisc, BackVisc);
+
+   /// Dispatch to the correct VertMix calculation
+   if (VertMixChoice == VertMixType::PP) {
+      LOG_INFO("VertMix:: into PP computeVertMix");
```

Comment:
These look like debug/test statements? We don't want to print stuff every time...